src/context.c

```diff
@@ -258,7 +258,7 @@ ly_ctx_new(const char *search_dir, uint16_t options, struct ly_ctx **new_ctx)
         search_dir_list = strdup(search_dir);
         LY_CHECK_ERR_GOTO(!search_dir_list, LOGMEM(NULL); rc = LY_EMEM, cleanup);
 
-        for (dir = search_dir_list; (sep = strchr(dir, ':')) != NULL && rc == LY_SUCCESS; dir = sep + 1) {
+        for (dir = search_dir_list; (sep = strchr(dir, PATH_SEPARATOR_CHAR)) != NULL && rc == LY_SUCCESS; dir = sep + 1) {
```

Comment:
Could just be `PATH_SEPARATOR_STR` renamed to `PATH_SEPARATOR` and here `PATH_SEPARATOR[0]` used.